ts/quickjs.test.ts

```diff
@@ -679,6 +685,67 @@ function contextTests(getContext: () => Promise<QuickJSContext>, isDebug = false
     })
   })
 
+  describe("intrisic objects", () => {
+    it("newBigInt - context respects intrinsic options", async () => {
+      const newContext = await getContext({ intrinsics: [] })
+
+      let handle;
+      try {
+        handle = newContext.newBigInt(2n ** 60n)
+        assert.fail('Must fail since BigInt is not an enabled intrinsic')
+      } catch (error) {
+        if (error instanceof QuickJSUnwrapError) {
+          assert.strictEqual(error.message, "not a function", "BigInt is not available without intrinsic object")
+        }
+      } finally {
+        if (handle?.alive) {
+          handle.dispose()
+        }
+        newContext.dispose()
+      }
+    })
+
+    it("evalCode - context respects intrinsic options", async () => {
+      // Eval is required to use `evalCode`
+      const newContext = await getContext({ intrinsics: ["Eval"] })
+
+      let handle;
+      try {
+        handle = newContext.unwrapResult(newContext.evalCode(`new Date()`))
+        assert.fail('Must fail since Date is not an enabled intrinsic')
+      } catch (e) {
+        if (e && typeof e === 'object' && 'name' in e) {
+          assert.equal(e.name, "RuntimeError")
+        } else {
+          assert.fail('Unknown error shape')
+        }
+      } finally {
+        if (handle?.alive) {
+          handle.dispose()
+        }
+        newContext.dispose()
+      }
+    })
+
+    it("evalCode - context executes as expected with default intrinsics", async () => {
+      // Eval is required to use `evalCode`
+      const newContext = await getContext({ intrinsics: [...DefaultIntrinsics] })
+
+      let handle;
+      try {
+        handle = newContext.unwrapResult(newContext.evalCode(`new Date()`))
+      } catch (e) {
+        // No error should occur
+        throw e
```

Comment:
@justjake When trying to add intrinsics to a context, an error gets throw here (`RuntimeError: memory access out of bounds`) from https://github.com/GrantMatejka/quickjs-emscripten/blob/26865efcebc281739fca84f8c3b368d2840106f4/ts/context.ts#L741

I'm not entirely sure what is causing this

Reply:
Does the stack trace for DEBUG build include any stacks on the C side? One of LLVM's sanitizers like address sanitizer might help point to the issue. Docs here: https://emscripten.org/docs/debugging/Sanitizers.html#address-sanitizer

Reply:
Here is the stack trace for the Debug build test
```
 DEBUG async module
         sync API
           intrisic objects
             evalCode - context executes as expected with default intrinsics:
     RuntimeError: memory access out of bounds
      at JS_CallInternal (wasm://wasm/005ac8be:wasm-function[112]:0x156dd)
      at JS_EvalFunctionInternal (wasm://wasm/005ac8be:wasm-function[317]:0x5f71a)
      at __JS_EvalInternal (wasm://wasm/005ac8be:wasm-function[135]:0x2c686)
      at JS_Eval (wasm://wasm/005ac8be:wasm-function[321]:0x61256)
      at QTS_Eval (wasm://wasm/005ac8be:wasm-function[72]:0xae29)
      at ret.<computed> (ts/generated/emscripten-module.WASM_DEBUG_ASYNCIFY.js:1646:35)
      at /home/grantmatejka/quickjs-emscripten/ts/generated/emscripten-module.WASM_DEBUG_ASYNCIFY.js:732:22
      at ccall (ts/generated/emscripten-module.WASM_DEBUG_ASYNCIFY.js:1895:22)
      at /home/grantmatejka/quickjs-emscripten/ts/generated/emscripten-module.WASM_DEBUG_ASYNCIFY.js:1934:16
      at QuickJSAsyncFFI.mustBeSync [as QTS_Eval] (ts/types-ffi.ts:108:20)
      at /home/grantmatejka/quickjs-emscripten/ts/context.ts:741:18
      at Lifetime.consume (ts/lifetime.ts:98:20)
      at QuickJSAsyncContext.evalCode (ts/context.ts:740:8)
      at Context.<anonymous> (ts/quickjs.test.ts:737:53)
```

Reply:
Here is the stack trace from the debug test case using default intrinsics
``` 
RuntimeError: memory access out of bounds
    at JS_GetGlobalVar (wasm://wasm/006b2e22:wasm-function[553]:0x5755d)
    at JS_CallInternal (wasm://wasm/006b2e22:wasm-function[405]:0x32940)
    at JS_CallFree (wasm://wasm/006b2e22:wasm-function[397]:0x2c739)
    at JS_EvalFunctionInternal (wasm://wasm/006b2e22:wasm-function[626]:0x68556)
    at __JS_EvalInternal (wasm://wasm/006b2e22:wasm-function[263]:0x18832)
    at JS_EvalInternal (wasm://wasm/006b2e22:wasm-function[631]:0x69c36)
    at JS_EvalThis (wasm://wasm/006b2e22:wasm-function[630]:0x69af7)
    at JS_Eval (wasm://wasm/006b2e22:wasm-function[632]:0x69cd8)
    at QTS_Eval (wasm://wasm/006b2e22:wasm-function[102]:0x6332)
    at /home/grantmatejka/quickjs-emscripten/ts/generated/emscripten-module.WASM_DEBUG_SYNC.js:728:22
    at ccall (/home/grantmatejka/quickjs-emscripten/ts/generated/emscripten-module.WASM_DEBUG_SYNC.js:4508:22)
    at QuickJSFFI.QTS_Eval (/home/grantmatejka/quickjs-emscripten/ts/generated/emscripten-module.WASM_DEBUG_SYNC.js:4525:16)
    at /home/grantmatejka/quickjs-emscripten/ts/context.ts:741:18
    at Lifetime.consume (/home/grantmatejka/quickjs-emscripten/ts/lifetime.ts:98:20)
    at QuickJSContext.evalCode (/home/grantmatejka/quickjs-emscripten/ts/context.ts:740:8)
    at Context.<anonymous> (/home/grantmatejka/quickjs-emscripten/ts/quickjs.test.ts:736:53)
Aborted(Assertion failed: list_empty(&rt->gc_obj_list), at: quickjs/quickjs.c,1983,JS_FreeRuntime)
        4) evalCode - context executes as expected with default intrinsics
```

Reply:
I'm pretty stumped on this and compiling with address sanitizer leads to a lot of `undefined symbol:` errors

Reply:
Disregard, but was that BaseObjects and BasicObjects are different 🤦 
BasicObjects are always attached to any context, but base objects need to be attached